bin/scripts/continious-data-population.js

```diff
@@ -0,0 +1,1861 @@
+                    writeMsg('INVALID POPULATOR_TEMPLATE_ID ', inputs[i].POPULATOR_TEMPLATE_ID);
+                }
+            }
+            if (!app[inputs[i].APP_ID]) {
+                db.close();
+                writeMsg('INVALID APP_ID ', inputs[i].APP_ID);
+            }
+            if (!template) {
+                db.close();
+                writeMsg('INVALID POPULATOR_TEMPLATE_ID ', inputs[i].POPULATOR_TEMPLATE_ID);
+            }
+            db.close();
+            users[inputs[i].APP_ID] = [];
+
+
+            pluginList[inputs[i].APP_ID] = await sendRequest({
+                requestType: 'GET',
+                Url: SERVER_URL + "/o/plugins?",
+                requestParamKeys: { app_id: inputs[i].APP_ID, api_key: API_KEY }
+            });
+        }
+    }
+    catch (error) {
+        clearInterval(interval);
+        writeMsg('There was an error while checking db parameters', error);
+    }
+}
+readDbParameters();
+
+var interval = setInterval(async function() {
+    for (var z = 0; z < Object.keys(inputs).length; z++) {
+        //users = [];
+        var userCount = getRandomInt(maxUserCount / 2, maxUserCount);
+        if (!users[inputs[z].APP_ID] || users[inputs[z].APP_ID].length === 0) {
+            for (var i = 0; i < userCount; i++) {
+                user = new getUser(template[inputs[z].APP_ID] && template[inputs[z].APP_ID].up);
+                users[inputs[z].APP_ID].push(user);
+            }
+        }
+        else {
+            if (users[inputs[z].APP_ID].length < maxUserCount) {
+                if (getRandomInt(1, 3) === 1) {
+                    user = new getUser(template && template.up);
+                    users[inputs[z].APP_ID].push(user);
+                }
+            }
+        }
+
+        var upperBoundary = Math.floor(userCount / 10);
+
+        var indexes = [];
+        for (var i0 = 0; i0 < upperBoundary; i0++) {
+            indexes.push(getRandomInt(0, users[inputs[z].APP_ID].length - 1));
+        }
+
+        indexes.forEach(function(index) {
+            if (users[inputs[z].APP_ID][index].hasSession) {
+                if (users[inputs[z].APP_ID][index].eventsInSession >= 1) {
+                    users[inputs[z].APP_ID][index].eventsInSession = 0;
+                    users[inputs[z].APP_ID][index].startSession(template[inputs[z].APP_ID], app[inputs[z].APP_ID], inputs[z].POPULATOR_TEMPLATE_ID);
+                }
+                else {
+                    users[inputs[z].APP_ID][index].addEvent(template[inputs[z].APP_ID], app[inputs[z].APP_ID], inputs[z].POPULATOR_TEMPLATE_ID);
+                    users[inputs[z].APP_ID][index].eventsInSession++;
+                }
+            }
+            else {
+                users[inputs[z].APP_ID][index].startSession(template[inputs[z].APP_ID], app[inputs[z].APP_ID], inputs[z].POPULATOR_TEMPLATE_ID);
+            }
+        });
+    }
+    if (intervalCount === 5) { // it should run one time each interval with some users.
```

Comment:
Why when intervalCount === 5?
Why can't we  do this at first and after we do that  setInterval()?

Reply:
In the populator, this code is called just before the populate is stopped. 
Current User and Today's Change fields in the cohort table are shown as 0 because the user is not created when I call it at the beginning.
I didn't want to do this after setInterval because it is not clear how long the script will run and the user will not be able to see the cohort as long as the script is running.
I thought it would be good to call this function after creating some users to show some numbers there.
But calling this function in the beginning would be fine also, after a certain time the computation for the cohort will work.